content/en/blog/_posts/2022-04-28-cni-version-related-breaking-change.md

```diff
@@ -0,0 +1,107 @@
+
+## An example containerd configuration file
+
+The following example shows a configuration for `containerd` runtime v1.6.x, which supports the latest version of CNI, v1.0.x.
+
+Please see the documentation from your plugin and networking provider for further instructions on configuring your system.
+
+By default, the `containerd` runtime adds a loopback interface, `lo`, to containers via the CNI plugin. The loopback plugin is distributed with the `containerd` release binaries that have the `cni` designation. `containerd` v1.6.0 and later includes a CNI v1.0.0-compatible loopback plugin as well as other default CNI plugins. The configuration for the loopback plugin is done internally and is set to CNI v1.0.0. This means the version of the loopback plugin must be v1.0.0 or later when this newer version `containerd` is started.
+
+The following bash command generates a default CNI config. Here, the 1.0.0 value for the config version is assigned to the `cniVersion` field for use when `containerd` invokes the CNI bridge plugin.
+
+```bash
+cat << EOF | tee /etc/cni/net.d/10-containerd-net.conflist
+{
+ "cniVersion": "1.0.0",
+ "name": "containerd-net",
+ "plugins": [
+   {
+     "type": "bridge",
+     "bridge": "cni0",
+     "isGateway": true,
+     "ipMasq": true,
+     "promiscMode": true,
+     "ipam": {
+       "type": "host-local",
+       "ranges": [
+         [{
+           "subnet": "10.88.0.0/16"
+         }],
+         [{
+           "subnet": "2001:4860:4860::/64"
+         }]
+       ],
+       "routes": [
+         { "dst": "0.0.0.0/0" },
+         { "dst": "::/0" }
+       ]
+     }
+   },
+   {
+     "type": "portmap",
+     "capabilities": {"portMappings": true}
+   }
+ ]
+}
+EOF
+```
+
+<a name="troubleshooting"></a>
+## Troubleshooting
+
+If the version of your CNI plugin does not correctly match the plugin version in the config because the config version is later than the plugin version, the containerd log may/will likely show an error message on startup of a pod similar to:
+
+```
+incompatible CNI versions; config is \"1.0.0\", plugin supports [\"0.1.0\" \"0.2.0\" \"0.3.0\" \"0.3.1\" \"0.4.0\"]"
+```
+
+Follow the instructions to update your CNI plugins and CNI config files to fix this problem.
+
+If the version of the plugin is missing in the CNI plugin config, the pod may run. However, when the pod is stopped, an error will be reported similar to:
+
+```
+ERRO[2022-04-26T00:43:24.518165483Z] StopPodSandbox for "b" failed                 error="failed to destroy network for sandbox \"bbc85f891eaf060c5a879e27bba9b6b06450210161dfdecfbb2732959fb6500a\": invalid version \"\": the version is empty"
```

Comment:
I'm not sure if this is the intended spacing of this error.

Reply:
Updated. Thanks.